packages/core/src/searchers/file-searcher.ts

```
@@ -32,26 +36,67 @@ export default class FileSearcher {
-    this.searchPromises = [];
   }
 
   /**
    * Invokes the search, for each search pattern.
    */
-  async search(): Promise<TaskItemData[]> {
+  async findFiles(): Promise<TaskItemData[]> {
     const resultData = [];
 
     for (const searchPatternName in this.searchPatterns) {
       if (Object.prototype.hasOwnProperty.call(this.searchPatterns, searchPatternName)) {
-        let data = await this._getSearchItem(searchPatternName);
+        let data = await this._getFileSearchItem(searchPatternName);
         resultData.push(toTaskItemData(searchPatternName, data));
       }
     }
     return resultData;
   }
 
-  _getSearchItem(searchPatternName: string): Promise<string[]> {
-    let patterns = this.searchPatterns[searchPatternName].concat(IGNORE_PATTERNS);
+  _getFileSearchItem(searchPatternName: string): Promise<string[]> {
+    let patterns = this.searchPatterns[searchPatternName].concat(
+      PATHS_TO_IGNORE.map((path) => `!${path}`)
+    );
     return globby(patterns, { cwd: this.baseDirectory });
   }
+
+  async findStrings(filesToSearch?: string[]): Promise<Record<string, number>> {
+    const files: string[] =
+      filesToSearch ||
+      walkSync(this.baseDirectory, {
+        ignore: PATHS_TO_IGNORE,
+      });
+
+    let countData: Record<string, number> = Object.keys(this.searchPatterns).reduce(
+      (a: Record<string, number>, b) => ((a[b] = 0), a),
+      {}
+    );
+
+    await Promise.all(
+      files.map((file) => {
+        return fs.promises
+          .readFile(path.join(path.resolve(this.baseDirectory), file), 'utf8')
+          .then((fileString: string) => {
+            for (let searchPatternName in this.searchPatterns) {
+              countData[searchPatternName] += this._getStringSearchCount(
+                fileString,
+                this.searchPatterns[searchPatternName]
+              );
+            }
+          })
+          .catch(() => {
+            debug(`Cant read file ${file}`);
```

Do we actually want to fail/notify here instead? Typically we use `debug` to output useful information when using that env var, such as a list of all the files we're searching. We wouldn't normally use it in a catch block.

---

hmm so generally what was being caught here was an attempt to read a dir, since walksync returns dirs, as well as plain files. so it would have `['foo/bar', 'foo/bar/blah.js']`. Should I just swallow these?

---

So what would you expect to happen if you were to execute this? Ignoring errors would lead to inconsistent results, so I think we should throw and ultimately error in Checkup. 

Perhaps this is a good example of Task error isolation - we can fail a single task, but shouldn't fail the whole checkup run. What do you think?

---

so i would say for tasks that are doing a string search on the app shouldn't necessarily fail if a single file cant be read.. i can filter the results from walksync so the _known_ catches will no longer happen, but regardless, string searching in this manner is pretty brittle, and if there is a file that cant be read in the repo, i dont know that we would want to throw an error. maybe we can add a count/list of files that count be parsed, but not actually fail the task?